v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/transforms/ProcessFailsafePubSubFn.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2021 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.elasticsearch.transforms;
+
+import com.google.cloud.teleport.v2.elasticsearch.templates.PubSubToElasticsearch;
+import com.google.cloud.teleport.v2.values.FailsafeElement;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.metrics.Counter;
+import org.apache.beam.sdk.metrics.Metrics;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.vendor.guava.v20_0.com.google.common.base.Throwables;
+
+/**
+ * The {@link ProcessFailsafePubSubFn} class processes a {@link FailsafeElement} containing an {@link PubsubMessage}
+ * and a String of the message's payload {@link PubsubMessage#getPayload()} into a {@link FailsafeElement} of
+ * the original {@link PubsubMessage} and a JSON string that has been processed with {@link Gson}.
+ * <p>
+ * If {@link PubsubMessage#getAttributeMap()} is not empty then the message attributes will be serialized along with
+ * the message payload.
+ */
+public class ProcessFailsafePubSubFn
```

Comment:
@prathapreddy123 @an2x 
This class is used once in [PubSubMessageToJsonDocument.java:82](https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/265/files#diff-e18d79bd4bf3cecb74413f670774f44522cc664acba07d86995a80917a6c7a8fR82). Let it be as a separate class or make it inner class of PubSubMessageToJsonDocument?

Reply:
Can be a Separate Class